bokeh/embed/bundle.py

```diff
@@ -223,13 +244,58 @@ def _bundle_extensions(objs, resources):
             continue
         names.add(name)
         module = __import__(name)
-        ext = ".min.js" if resources is None or resources.minified else ".js"
-        artifact = join(dirname(module.__file__), "dist", name + ext)
-        if exists(artifact):
-            bundle = BaseResources._inline(artifact)
-            extensions.append(bundle)
+        this_file = abspath(module.__file__)
+        base_dir = dirname(this_file)
+        dist_dir = join(base_dir, "dist")
+
+        ext_path = join(base_dir, "bokeh.ext.json")
+        if not exists(ext_path):
+            continue
 
-    return extensions
+        server_prefix = f"{resources.root_url}static/extensions"
+        package_path = join(base_dir, "package.json")
+
+        pkg: Optional[str] = None
+
+        if exists(package_path):
+            with open(package_path) as io:
+                try:
+                    pkg = json.load(io)
+                except json.decoder.JSONDecodeError:
+                    pass
+
+        artifact_path: str
+        server_url: str
+        cdn_url: Optional[str] = None
+
+        if pkg is not None:
+            pkg_name = pkg["name"]
+            pkg_version = pkg.get("version", "latest")
+            pkg_main = pkg.get("module", pkg.get("main", None))
+            if pkg_main is not None:
+                cdn_url = f"https://unpkg.com/{pkg_name}@^{pkg_version}/{pkg_main}"
```

Comment:
can we make the CDN base URL configurable? I can imagine folks wanting to use different ones (e.g I've never heard of this one, but have heard cdnjs mentioned often)

Reply:
At some point. CDN handling is a secondary concern of this PR.

Reply:
At the very least pull it in to a module level attr so that there is not an arbitrary constant value buried in code.